docs/asciidoc/static/include/pluginbody.asciidoc

```diff
@@ -0,0 +1,1021 @@
+
+There are lots more examples in the https://github.com/logstash-plugins?query=logstash-input[logstash-plugin github repository].
+
+endif::run_method[]
+
+// /////////////////////////////////////////////////////////////////////////////
+// If receive_method is defined (should only be for output plugin page)
+// /////////////////////////////////////////////////////////////////////////////
+ifdef::receive_method[]
+==== The `receive` Method
+[source,ruby]
+[subs="attributes"]
+----------------------------------
+  public
+  def receive(event)
+  end # def event
+----------------------------------
+This empty `receive` method does absolutely nothing, which is valid code from
+the https://github.com/logstash-plugins/logstash-output-null/blob/master/lib/logstash/outputs/null.rb[null]
+output plugin.
+
+The `receive` method _receives_ events from the pipeline and further processes
+them before sending them to their final destination.  These destinations can
+be as varied as https://github.com/logstash-plugins/logstash-output-file/blob/master/lib/logstash/outputs/file.rb[files],
+https://github.com/logstash-plugins/logstash-output-elasticsearch/blob/master/lib/logstash/outputs/elasticsearch.rb[elasticsearch],
+https://github.com/logstash-plugins/logstash-output-email/blob/master/lib/logstash/outputs/email.rb[email],
+https://github.com/logstash-plugins/logstash-output-rabbitmq/blob/master/lib/logstash/outputs/rabbitmq.rb[RabbitMQ],
+https://github.com/logstash-plugins/logstash-output-tcp/blob/master/lib/logstash/outputs/tcp.rb[tcp],
+https://github.com/logstash-plugins/logstash-output-stdout/blob/master/lib/logstash/outputs/stdout.rb[stdout],
+and dozens more!
+
+Output plugins do not require you to make use of codecs, but you can use a
+codec by including code similar to this:
+[source,ruby]
+[subs="attributes"]
+----------------------------------
+  public
+  def receive(event)
+    return unless output?(event)
+    @codec.encode(event)
+  end # def receive
+----------------------------------
+
+For more examples of output plugins, see the https://github.com/logstash-plugins?query=logstash-output[logstash-plugin github repository].
+
+endif::receive_method[]
+
+// /////////////////////////////////////////////////////////////////////////////
+// If teardown_method is defined (should only be for input or output plugin page)
+// /////////////////////////////////////////////////////////////////////////////
+ifdef::teardown_method[]
+==== The `teardown` Method
+[source,ruby]
+[subs="attributes"]
+----------------------------------
+  public
+  def teardown
+    @udp.close if @udp && !@udp.closed?
+  end
+----------------------------------
+The `teardown` method is not present in all input or output plugins.  It is
+called when a shutdown happens to ensure that sockets, files, connections and
+threads are all closed down properly.  If your plugin uses these connections,
+you should include a teardown method.
+endif::teardown_method[]
+
+=== Add a Gemfile
```

Comment:
Again, the overall hierarchy is getting fuzzy, but this seems like it should be part of an overall process that's it's kind of lost touch with. Like "Create the class file, add a gemfile, add a gemsec file?"